linkerd/http-classify/src/service.rs

```diff
@@ -48,3 +49,24 @@ where
         self.inner.proxy(svc, req)
     }
 }
+
+impl<C, S, B> Service<http::Request<B>> for Classify<C, S>
+where
+    C: super::Classify,
+    S: tower::Service<http::Request<B>>,
+{
+    type Response = S::Response;
+    type Error = S::Error;
+    type Future = S::Future;
+
+    #[inline]
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        self.inner.poll_ready(cx)
+    }
+
+    fn call(&mut self, mut req: http::Request<B>) -> Self::Future {
+        let classify_rsp = self.classify.classify(&req);
+        let _ = req.extensions_mut().insert(classify_rsp);
```

Comment:
nit: should this `debug_assert!` that there wasn't already a response classifier inserted? not a big deal.